File: java/src/merge_sort/randomIntTestData.java
package merge_sort;

import java.util.Arrays;
import java.util.Random;

class randomIntTestData {
	private static final Random rand = new Random(0);

	public int[] array, expectedArray;

	randomIntTestData() {
		int n = rand.nextInt(50), max = rand.nextInt(50);
		array = new int[n];
		for (int i = 0; i < n; i++) {
			array[i] = rand.nextInt(1 + max);
		}
		expectedArray = array.clone();
		Arrays.sort(expectedArray);
	}
}

File: go/merge_sort/random_int_test_data.go
package merge_sort

import (
	"math/rand/v2"
	"sort"
)

type randomIntTestData struct {
	array, expectedArray []int
}

var random = rand.New(rand.NewPCG(0, 0))

func newRandomIntTestData() *randomIntTestData {
	var td randomIntTestData

	n, maxVal := random.IntN(50), random.IntN(50)
	td.array = make([]int, n)
	for i := range n {
		td.array[i] = random.IntN(1 + maxVal)
	}
	td.expectedArray = append([]int{}, td.array...)
	sort.Ints(td.expectedArray)

	return &td
}
